packages/nextjs/src/utils/config.ts

```diff
@@ -3,33 +3,36 @@ import { logger } from '@sentry/utils';
 import defaultWebpackPlugin, { SentryCliPluginOptions } from '@sentry/webpack-plugin';
 import * as SentryWebpackPlugin from '@sentry/webpack-plugin';
 
+const SENTRY_CLIENT_CONFIG_FILE = './sentry.client.config.js';
+const SENTRY_SERVER_CONFIG_FILE = './sentry.server.config.js';
+// this is where the transpiled/bundled version of `USER_SERVER_CONFIG_FILE` will end up
+export const SERVER_INIT_OUTPUT_LOCATION = 'sentry/serverInit';
+
 // eslint-disable-next-line @typescript-eslint/no-explicit-any
 type PlainObject<T = any> = { [key: string]: T };
 
-// Man are these types hard to name well. "Entry" = an item in some collection of items, but in our case, one of the
-// things we're worried about here is property (entry) in an object called... entry. So henceforth, the specific
-// property we're modifying is going to be known as an EntryProperty.
-
 // The function which is ultimately going to be exported from `next.config.js` under the name `webpack`
 type WebpackExport = (config: WebpackConfig, options: WebpackOptions) => WebpackConfig;
 
 // The two arguments passed to the exported `webpack` function, as well as the thing it returns
-type WebpackConfig = { devtool: string; plugins: PlainObject[]; entry: EntryProperty };
+type WebpackConfig = { devtool: string; plugins: PlainObject[]; entry: EntryProperty; output: { path: string } };
 type WebpackOptions = { dev: boolean; isServer: boolean; buildId: string };
 
 // For our purposes, the value for `entry` is either an object, or a function which returns such an object
 type EntryProperty = (() => Promise<EntryPropertyObject>) | EntryPropertyObject;
-
 // Each value in that object is either a string representing a single entry point, an array of such strings, or an
 // object containing either of those, along with other configuration options. In that third case, the entry point(s) are
 // listed under the key `import`.
-type EntryPropertyObject = PlainObject<string | Array<string> | EntryPointObject>;
+type EntryPropertyObject = PlainObject<string> | PlainObject<Array<string>> | PlainObject<EntryPointObject>;
```

Comment:
I think this should work as well, but you'd have to confirm whether it allows for the same property-based `for in/of` as `[key: string]` does. If so, then `PlainObject` can be removed entirely.

Reply:
I thought I'd learned that `Record` is really meant to be used on a known, finite set of keys. (I get that it *would* work - I'm talking about *should*.) Is that not true? 

I made `PlainObject` just because I was tired of typing `{[key: string] : whatever} a million times, and figured since there's `Array` their might as well be an equivalent for objects. ¯\_(ツ)_/¯ I'm not committed to it.

Reply:
> Is that not true? 

Nope, it's basically what you wrote up there with key/value.

https://www.typescriptlang.org/play?ssl=12&ssc=1&pln=13&pc=1#code/C4TwDgpgBAYg9nKBeKAlCBjOAnAJgHgGdhsBLAOwHMAaKcgVwFsAjCbAPgG4Aobrc4lABmCAFywEyKAG9uUKAAtS4gIzU5UYArYRxAJnXzccSpTjiAzNwC+vEXAB0GAIbBgiFABYe9gNoByY0oIfwBdKQBWHm4RbCgACn5BAGsIECgKYQQAShkNJLgAGwgHQpN41JBaP0rQ7JteJOAoSql-JX8eAuLS8pq0us4gA

Reply:
Yeah, no, I know it works. I just meant idiomatically. Here are a few things I dug out of my bookmarks:

https://stackoverflow.com/a/54100359/3874926

https://fnune.com/typescript/2019/01/30/typescript-series-1-record-is-usually-not-the-best-choice

https://rickcarlino.com/2017/02/27/real-world-use-case-for-typescript-record-types.html

https://stackoverflow.com/questions/51936369/what-is-the-record-type-in-typescript#comment103989203_51937036

Happy to go back to using `{ [key: string]: string }` if you think it's better than `PlainObject`.

UPDATE: The playground hates it, but check out what compiles just fine if you actually use it in your code:

```
interface Object<T> extends ReturnType<ObjectConstructor> {
  [key: string]: T;
}

const numObject: Object<number> = {
  someNum: 4,
  someNonNum: 'whoops', // Error: Type 'string' is not assignable to type 'number'.
};
```

I get that it's a little unorthodox, but what do you think of using that?? 😁